drivers/sensor/adxl345/adxl345.c

```diff
@@ -91,7 +91,7 @@ static int adxl345_channel_get(struct device *dev,
 {
 	struct adxl345_dev_data *data = dev->driver_data;
 
-	if (data->sample_number > 32) {
+	if (data->sample_number > (ADXL345_MAX_FIFO_SIZE - 1)) {
```

Comment:
This should be:

    if (data->sample_number >= ARRAY_SIZE(data->bufx)) {
        data->sample_number = 0;
    }

to avoid dependencies on how the array is actually defined.

This should also be changed in the `__ASSERT_NO_MSG` in `adxl345_sample_fetch`.

Reply:
Fixed and split into two commits.